shared/router-v2/router.shared.tsx

```diff
@@ -54,10 +54,13 @@ const useDarkNeedsRedraw = () => {
 
 const useNavKey = (appState: AppState, key: React.MutableRefObject<number>) => {
   const needsRedraw = useDarkNeedsRedraw()
-  if (needsRedraw) {
-    key.current++
+  if (appState === AppState.NEEDS_INIT) {
+    key.current = -1
+  } else {
+    if (needsRedraw) {
+      key.current++
+    }
   }
-  return appState === AppState.NEEDS_INIT ? -1 : key.current
```

Comment:
we return and use -1 but then don't update the ref so it'll thrash when we recall this hook and appState changes, so instead just update the ref always